libyul/ObjectOptimizer.cpp

```diff
@@ -154,12 +154,12 @@ std::optional<h256> ObjectOptimizer::calculateCacheKey(
 	rawKey += keccak256(asmPrinter(_ast)).asBytes();
 	rawKey += keccak256(_debugData.formatUseSrcComment()).asBytes();
 	rawKey += h256(u256(_settings.language)).asBytes();
-	rawKey += FixedHash<1>(uint8_t(_settings.optimizeStackAllocation ? 0 : 1)).asBytes();
+	rawKey += FixedHash<1>(uint8_t(_settings.optimizeStackAllocation ? 1 : 0)).asBytes();
```

Comment:
It's fine now (and also doesn't matter in the sense of clashing hashes) but I do think a static cast to uint8 would have been better than the ternary.

Reply:
I wanted to answer that `bool` can have values other than `0` and `1` (even if no one *should* ever do that in practice) so the cast is technically incorrect. But then I found out that I was wrong. I did not expect that, but it makes perfect sense given that it's `static_cast<>` and not `reinterpret_cast<>`:

```c++
static_assert(static_cast<uint8_t>(static_cast<bool>(2)) == 1);
```

TIL.

And just another reason why it's better to avoid the C-style casts. Not sure why I even used one here, it should have been a static cast from the beginning :)

Reply:
Yup - I was confused at first why you kept the ternaries but thought oh well, ternaries plus static cast is still better than c-style cast. :)

Reply:
And actually I think bool being (on a binary level) something else than 0 or 1 leads to UB